File: 1291.py
class Solution:
    def sequentialDigits(self, low: int, high: int) -> List[int]:
        lLen = len(str(low))
        hLen = len(str(high))
        res = list()
        return [
            val
            for l in range(lLen, hLen + 1)
            for d in range(1, 9 - l + 2)
            if low <= (val := int(''.join(str(i) for i in range(d, l + d)))) <= high
        ]

File: sol/1291/1291.cpp
struct Solution {
    static const string dseq;

    vector<int> sequentialDigits(int low, int high) {
        vector<int> res;
        int ll = to_string(low).size(), hl = to_string(high).size(), val;
        for (int l = ll; l <= hl; ++l){
            for (int s = 0; s < 10 - l; ++s){
                val = stoi(this->dseq.substr(s, l));
                if (low <= val && val <= high){
                    res.push_back(val);
                }
            }
        }
        return res;
    }
};
const string Solution::dseq = "123456789";
